systemd/systemd.go

```diff
@@ -513,8 +536,67 @@ Options=%s
 
 [Install]
 WantedBy=multi-user.target
-`, name, revision, what, where, fstype, strings.Join(options, ","))
+`, snapName, revision, what, where, fstype, strings.Join(options, ","))
 
 	mu := MountUnitPath(where)
-	return filepath.Base(mu), osutil.AtomicWriteFile(mu, []byte(c), 0644, 0)
+	mountUnitName, err := filepath.Base(mu), osutil.AtomicWriteFile(mu, []byte(c), 0644, 0)
+	if err != nil {
+		return "", err
+	}
+
+	// we need to do a daemon-reload here to ensure that systemd really
+	// knows about this new mount unit file
+	if err := s.daemonReloadNoLock(); err != nil {
+		return "", err
+	}
+
+	if err := s.Enable(mountUnitName); err != nil {
+		return "", err
+	}
+	if err := s.Start(mountUnitName); err != nil {
+		return "", err
+	}
+
+	return mountUnitName, nil
+}
+
+func (s *systemd) RemoveMountUnitFile(mountedDir string) error {
+	daemonReloadLock.Lock()
+	defer daemonReloadLock.Unlock()
+
+	unit := MountUnitPath(dirs.StripRootDir(mountedDir))
+	if !osutil.FileExists(unit) {
+		return nil
+	}
+
+	// use umount -d (cleanup loopback devices) -l (lazy) to ensure that even busy mount points
+	// can be unmounted.
+	// note that the long option --lazy is not supported on trusty.
+	// the explicit -d is only needed on trusty.
+	isMounted, err := osutil.IsMounted(mountedDir)
```

Comment:
While this is not new logic I would like to understand the motivation behind this particular arrangement of actions:
- we unmount the snap ourselves (MNT_DETACH)
- we stop the mount unit (which we could have by using LazyUnmount=true in the mount unit)
- we disable the mount unit (what for?)
- we remove the mount unit
- we reload systemd

Why is the logic simply not:

- we disable --now the mount unit via systemctl
- we remove the mount unit by removing the file
- we reload systemd

Reply:
does trusty have a working disable --now?